_build_meta.py

```diff
@@ -13,6 +13,78 @@
 We comply by re-publishing setuptools' ``build_meta``.
 """
 
+import sys
+from pathlib import Path
+
+if sys.version_info > (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+from importlib.metadata import Distribution, DistributionFinder
+
 from setuptools import build_meta
 
+
+def _entry_points_txt():
+    """
+    Format the entry points from ``pyproject.toml`` as the INI-style
+    ``entry_points.txt`` used in package metadata.
+    """
+    with open(Path(__file__).parent / "pyproject.toml", "rb") as f:
+        data = tomllib.load(f)
+
+    lines = []
+    for section, pairs in data["project"]["entry-points"].items():
+        lines.append(f"[{section}]")
+        for key, value in pairs.items():
+            lines.append(f"{key} = {value}")
+        lines.append("")
+    return "\n".join(lines)
```

Comment:
This docstring is no longer accurate with all this new code that now lives in here. I don't quite understand why this needs to happen in this module?

Reply:
Whoops, I'm sorry, I meant to revert all this.

The answer to your question is that this allows setuptools to find Incremental's entry point when running as a local backend (the `backend-path` stuff in `pyproject.toml`). In that context the incremental package hasn't been installed, so the entry point metadata doesn't exist. This code injects the entry points at runtime.

This is a silly amount of code and complexity for what it does. The old `[tool.setuptools.dynamic]` hack was more pragmatic.